src/test/java/com/amazon/opendistroforelasticsearch/security/dlic/rest/api/AuditApiActionTest.java

```diff
@@ -52,6 +69,177 @@ public void testInvalidPath() throws Exception {
+        final List<String> readonlyFields = ImmutableList.of("/audit/enable_rest", "/audit/disabled_rest_categories", "/audit/ignore_requests", "/compliance/read_watched_fields");
+
+        final Settings settings = Settings.builder()
+                .putList(ConfigConstants.OPENDISTRO_SECURITY_AUDIT_CONFIG_READONLY_FIELDS, readonlyFields)
+                .build();
+        setupWithRestRoles(settings);
+        final ObjectMapper objectMapper = DefaultObjectMapper.objectMapper;
+
+        // test get
+        RestHelper.HttpResponse response = rh.executeGetRequest(ENDPOINT, adminCredsHeader);
+        assertEquals(HttpStatus.SC_OK, response.getStatusCode());
+        List<String> actual = Streams.stream(readTree(response.getBody()).get("_readonly").iterator())
+                .map(JsonNode::textValue)
+                .collect(Collectors.toList());
+        assertEquals(readonlyFields, actual);
+
+        // test config
+        final AuditConfig auditConfig = AuditConfig.from(Settings.EMPTY);
+
+        // reset
+        ObjectNode json = objectMapper.valueToTree(auditConfig);
+        testPutRequest(json, HttpStatus.SC_OK, true);
+        // change enable_rest readonly property
+        testReadonlyBoolean(json, "/audit", "enable_rest");
+
+        // reset
+        json = objectMapper.valueToTree(auditConfig);
+        testPutRequest(json, HttpStatus.SC_OK, true);
+        // change disabled_rest_categories readonly property
+        testReadonlyCategories(json, "/audit", "disabled_rest_categories");
+
+        // reset
+        json = objectMapper.valueToTree(auditConfig);
+        testPutRequest(json, HttpStatus.SC_OK, true);
+        // change ignore_requests readonly property
+        testReadonlyList(json, "/audit", "ignore_requests");
+
+        // reset
+        json = objectMapper.valueToTree(auditConfig);
+        testPutRequest(json, HttpStatus.SC_OK, true);
+        // change ignore_requests readonly property
+        testReadonlyMap(json, "/compliance", "read_watched_fields");
+    }
+
+    private void testPutRequest(final JsonNode json, final int expectedStatus, final boolean sendAdminCertificate, final Header... header) throws Exception {
+        rh.sendAdminCertificate = sendAdminCertificate;
+        RestHelper.HttpResponse response = rh.executePutRequest(CONFIG_ENDPOINT, writeValueAsString(json, false), header);
+        assertEquals(expectedStatus, response.getStatusCode());
+    }
+
+    private void testReadonlyBoolean(final ObjectNode json, final String config, final String resource) throws Exception {
+        final String resourcePath = "/config" + config + "/" + resource;
+        ((ObjectNode)json.at(config)).put(resource, true);
+        testPutRequest(json, HttpStatus.SC_OK, true);
+        ((ObjectNode)json.at(config)).put(resource, false);
+        testPutRequest(json, HttpStatus.SC_CONFLICT, false, adminCredsHeader);
+        testBooleanPatch(resourcePath, false, HttpStatus.SC_CONFLICT, adminCredsHeader);
```

Comment:
Should we similarly test with admin certificate in the additional Patch methods ?

Reply:
I have added on line 146 to test all actions as super-admin which includes Patch methods